sqlmesh/core/scheduler.py

```diff
@@ -549,19 +558,33 @@ def compute_interval_params(
         end_bounded=end_bounded,
     ).items():
         if signal_factory and snapshot.is_model:
+            unready = set(intervals)
+
             for signal in snapshot.model.render_signals(
                 start=start, end=end, execution_time=execution_time
             ):
                 intervals = _check_ready_intervals(
                     signal=signal_factory(signal),
                     intervals=intervals,
                 )
+            unready -= set(intervals)
+        else:
+            unready = set()
+
+        for parent in snapshot.model.depends_on:
+            if parent in all_unready_intervals:
+                unready.update(all_unready_intervals[parent])
```

Comment:
The batches can be completely misaligned between the parent and the child. It's not just cron granularity but also a different batch size.

Reply:
Never mind the batch_size. It comes in later